From 5e09e5b5e12c240305baf10fa5c961445277b0a7 Mon Sep 17 00:00:00 2001
From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Date: Mon, 17 Mar 2014 09:05:17 -0700
Subject: [PATCH] Don't skip the current cdev

When multiple sources are using same cooling device, we don't apply
compensation without giving some time to previous activation to have
some affect. But continuing loop cause next cdev in order to start
activation. But instead let the previous device complete it action.
---
 src/thd_trip_point.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/thd_trip_point.cpp b/src/thd_trip_point.cpp
index 6704160..0cead81 100644
--- a/src/thd_trip_point.cpp
+++ b/src/thd_trip_point.cpp
@@ -135,7 +135,7 @@ bool cthd_trip_point::thd_trip_point_check(int id, unsigned int read_temp,
 					thd_log_info("Too early to act index %d tm %ld\n",
 							cdev->thd_cdev_get_index(),
 							tm - cdevs[i].last_op_time);
-					continue;
+					break;
 				}
 				cdevs[i].last_op_time = tm;
 			}
-- 
1.9.0

